app/models/deploy_service.rb

```diff
@@ -19,7 +19,13 @@ def deploy!(stage, attributes)
   def confirm_deploy!(deploy)
     send_before_notifications(deploy)
 
-    job_execution = JobExecution.start_job(deploy.reference, deploy.job)
+    stage = deploy.stage
+
+    job_execution = JobExecution.start_job(
+      deploy.reference, deploy.job,
+      construct_env(stage).merge(key: stage.id)
+    )
+
     send_sse_deploy_update('start', deploy)
 
     job_execution.on_complete do
```

Comment:
I just noticed there's a race condition here.  The job actually gets started when you call `start_job` above, which spawns off a thread.  Here in the code it's adding a callback via `on_complete`, but just assuming that the JobExecution hasn't gotten to that point yet.